app/src/main/java/org/kiwix/kiwixmobile/utils/LanguageUtils.java

```diff
@@ -340,4 +340,17 @@ public String getLanguageName() {
       return mLanguageName;
     }
   }
+
+  public static String getStringOrResource(Context appContext, String str){
```

Comment:
The name's a bit vague but I can't think of one that'd be clearer at the moment. This method aims to obtain a localised string when asked to do so. It does two things in my view: First it checks whether the provided string contains a special string. (Secondly) If so it requests a localised string that matches the rest of the contents of this string. Otherwise it returns the string as-is.

I suggest it's worth splitting the method into two. One determines whether to ask for a localised resource, the other tries to obtain the localised resource.

This is also vulnerable to the string not being available e.g. if someone asks for REPLACE_this_isnt_available. What would you like the code to do? What would users like the app to do? 

How about writing some tests that exemplify what you'd like the code to do in various circumstances?

Reply:
I've restructured some code, but testing is a bit trickier because tests don't have `getApplicationContext()`.
May be worth further discussion about this?